analysis_options.yaml

```diff
@@ -1,2 +1,215 @@
+# Updated according to
+# https://dart-lang.github.io/linter/lints/index.html
+# at 2020-08-10
+
 include: package:pedantic/analysis_options.yaml
 
+analyzer:
+  enable-experiment:
+    - non-nullable
+
+  strong-mode:
+    implicit-casts: false
+# TODO(zoechi) conflicts with some other rules and is subject to be reworked
+# https://github.com/dart-lang/sdk/issues/33749
+# but in general a useful setting to be enabled eventually
+#    implicit-dynamic: false
+
+#  language:
+# TODO(zoechi) check if this is already available
+# https://github.com/dart-lang/language/blob/master/resources/type-system/strict-raw-types.md
+#    strict-raw-types: true
+
+linter:
+  rules:
+## Error Rules
+
+# personal preference this conflicts with prefer_relative_imports
+#    - always_use_package_imports
+    - avoid_empty_else
+    - avoid_print
+    - avoid_relative_lib_imports
+    - avoid_returning_null_for_future
+    - avoid_slow_async_io
+    - avoid_types_as_parameter_names
+    - avoid_web_libraries_in_flutter
+    # subject to false positives, perhaps worth supressing in code in such cases
+    - cancel_subscriptions
+    # subject to false positives, perhaps worth supressing in code in such cases
+    - close_sinks
+    - comment_references
+    - control_flow_in_finally
+    - diagnostic_describe_all_properties
+    - empty_statements
+    - hash_and_equals
+    - invariant_booleans
+    - iterable_contains_unrelated_type
+    - list_remove_unrelated_type
+    - literal_only_boolean_expressions
+    - no_adjacent_strings_in_list
+    - no_duplicate_case_values
+    - no_logic_in_create_state
+    - prefer_relative_imports      # personal preference
+    - prefer_void_to_null
+    - test_types_in_equals
+    - throw_in_finally
+    - unnecessary_statements
+    - unrelated_type_equality_checks
+    - unsafe_html
+    - use_key_in_widget_constructors
```

Comment:
Will flutter specific rules be valid here?

Reply:
As far as I know they don't interfere but won't have any effect.
I'll remove it.

Reply:
Removed both `unsafe_html` and `use_key_in_widget_constructors`